CSSOM updates to match web API templates (#44993)
mdn/content: files/en-us/web/api/css_typed_om_api/index.md

CSS Typed OM both allows for the performant manipulation of values assigned to CSS properties while enabling maintainable code that is both more understandable and easier to write.

## Guides

- [Using the CSS Typed Object Model](/en-US/docs/Web/API/CSS_Typed_OM_API/Guide)
  - : An introduction to the main features of the typed object model.

## Interfaces

- {{domxref('CSSImageValue')}}
  - : Represents values for properties that take an image, such as {{cssxref("background-image")}}, {{cssxref("list-style-image")}}, or {{cssxref("border-image-source")}}.
- {{domxref('CSSKeywordValue')}}

>   - : Represent the value of a CSS keyword or other identifier, such as `initial`, or `auto`.

Reviewer: ```suggestion
  - : Represents the value of a CSS keyword or other identifier, such as `initial`, or `auto`.
```

    When used where a string is expected, it will return the value of `CSSKeyword.value`.
- {{domxref('CSSMathClamp')}}
  - : Represents the CSS {{cssxref("clamp","clamp()")}} function.
- {{domxref('CSSMathInvert')}}

>   - : represents a CSS {{cssxref("calc","calc()")}} value used as `calc(1 / <value>)`. This type is used internally by {{domxref('CSSNumericValue.div','div()')}}, to create an appropriate {{domxref('CSSMathProduct')}}.

Reviewer: ```suggestion
  - : Represents a CSS {{cssxref("calc","calc()")}} value used as `calc(1 / <value>)`. This type is used internally by {{domxref('CSSNumericValue.div','div()')}}, to create an appropriate {{domxref('CSSMathProduct')}}.
```

(and lots more like this)

- {{domxref('CSSMathMax')}}
  - : represents the CSS {{cssxref("max","max()")}} function.
- {{domxref('CSSMathMin')}}
  - : represents the CSS {{cssxref("min","min()")}} function.
- {{domxref('CSSMathNegate')}}
  - : negates the value passed into it.
- {{domxref('CSSMathProduct')}}
  - : represents the result obtained by calling {{domxref('CSSNumericValue.mul','mul()')}} or {{domxref('CSSNumericValue.div','div()')}} on {{domxref('CSSNumericValue')}}.
- {{domxref('CSSMathSum')}}
  - : represents the result obtained by calling {{domxref('CSSNumericValue.add','add()')}}, {{domxref('CSSNumericValue.sub','sub()')}}, or {{domxref('CSSNumericValue.toSum','toSum()')}} on {{domxref('CSSNumericValue')}}.
- {{domxref('CSSMathValue')}}
  - : The base class for numeric values that are more complicated than a single value and unit, such as `CSSMathMax` and `CSSMathSum`
- {{domxref('CSSMatrixComponent')}}

>   - : TBD

Reviewer: Will these have content? I don't think we can leave it like this.

- {{domxref('CSSNumericArray')}}
  - : TBD
- {{domxref('CSSNumericValue')}}

>   - : An interface representing operations that all numeric values can perform, such as addition, multiplication, and so on.

Reviewer: ```suggestion
  - : Represents operations that all numeric values can perform, such as addition, multiplication, and so on.
```

(and similar. We should be consistent here.)

- {{domxref('CSSPerspective')}}
  - : TBD
- {{domxref('CSSPositionValue')}}

>   - : Represents values for properties that take a position, for example object-position.

Reviewer: Link object-position.

>   - : Represents values for properties that take a position, for example object-position.

Reviewer: ```suggestion
  - : Represents values for properties that take a position, such as object-position.
```

- {{domxref('CSSRotate')}}
  - : TBD
- {{domxref('CSSScale')}}
  - : TBD
- {{domxref('CSSSkew')}}
  - : TBD
- {{domxref('CSSSkewX')}}
  - : TBD
- {{domxref('CSSSkewY')}}
  - : TBD
- {{domxref('CSSStyleValue')}}

>   - : Base class of all CSS values accessible through the Typed OM API.

Reviewer: ```suggestion
  - : Base class for all CSS values accessible through the Typed OM API.
```

(and we say "The base class" above)

    An instance of this class may be used anywhere a string is expected.
- {{domxref('CSSTransformComponent')}}
  - : TBD
- {{domxref('CSSTransformValue')}}

>   - : An interface representing a list of {{cssxref("transform")}} list values. They "contain" one or more {{domxref('CSSTransformComponent')}}s, which represent individual `transform` function values.

Reviewer: ```suggestion
  - : Represents a list of {{cssxref("transform")}} list values. Instances of this interface "contain" one or more {{domxref('CSSTransformComponent')}} instances, which each represent individual `transform` function values.
```

- {{domxref('CSSTranslate')}}
  - : TBD
- {{domxref('CSSUnitValue')}}

>   - : An interface representing numeric values that can be represented as a single unit, or a named number and percentage.

Reviewer: ```suggestion
  - : Represents numeric values that can be represented as a single unit, or a named number and percentage.
```

etc etc

- {{domxref('CSSUnparsedValue')}}

>   - : Represents property values that reference [custom properties](/en-US/docs/Web/CSS/Reference/Properties/--*). It consists of a list of string fragments and variable references. Represents property values that reference custom properties. It consists of a list of string fragments and variable references.

Reviewer: ```suggestion
  - : Represents property values that reference [custom properties](/en-US/docs/Web/CSS/Reference/Properties/--*). Instances of this interface consist of a list of string fragments and variable references.
```

(not exactly sure what "list of string fragments and variable references" means)

- {{domxref('CSSVariableReferenceValue')}}
  - : TBD
- {{domxref('StylePropertyMap')}}
  - : The {{domxref('StylePropertyMap')}} interface of the CSS Typed Object Model API provides a representation of a CSS declaration block that is an alternative to `CSSStyleDeclaration`.
- {{domxref('StylePropertyMapReadOnly')}}
  - : TBD

> 

Reviewer: This section should also include an [Extensions to other interfaces](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/API_landing_page_template#extensions_to_other_interfaces) section listing APIs like [computedStyleMap()](https://developer.mozilla.org/en-US/docs/Web/API/Element/computedStyleMap), [attributeStyleMap](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/attributeStyleMap) etc.

> ## Specifications

Reviewer: Why list the spec four times?
